lib/widgets/scrollbar.dart

```diff
@@ -122,474 +109,69 @@ class _AppScrollbarState extends State<AppScrollbar> {
     final highlightColor = theme.highlightColor;
     return Theme(
       data: theme.copyWith(
-        highlightColor: ThemeControl.instance.isDark ? const Color(0x40CCCCCC) : const Color(0x66BCBCBC),
-      ),
+          highlightColor: ThemeControl.instance.isDark ? const Color(0x40CCCCCC) : const Color(0x66BCBCBC),
+          scrollbarTheme: widget.hoverThickness == null
+              ? theme.scrollbarTheme
+              : theme.scrollbarTheme.copyWith(thickness: MaterialStateProperty.all(widget.hoverThickness))),
       child: MediaQuery.removePadding(
         context: context,
         removeTop: true,
-        child: Stack(
-          children: [
-            _Scrollbar(
-              controller: controller,
-              thumbVisibility: widget.thumbVisibility,
-              showTrackOnHover: widget.showTrackOnHover,
-              hoverThickness: widget.hoverThickness,
-              thickness: widget.thickness,
-              radius: widget.radius,
-              notificationPredicate: widget.notificationPredicate,
-              interactive: widget.interactive,
-              onThumbPressStart: () => setState(() {
-                _dragIsActive = true;
-              }),
-              onThumbPressEnd: () => setState(() {
-                _dragIsActive = false;
-              }),
-              child: Theme(
-                data: theme.copyWith(highlightColor: highlightColor),
-                child: widget.child,
+        child: NotificationListener<ScrollNotification>(
+          onNotification: (notification) {
+            final Offset? dragPosition;
+            if (notification is ScrollStartNotification) {
+              dragPosition = notification.dragDetails?.localPosition;
+            } else if (notification is ScrollUpdateNotification) {
+              dragPosition = notification.dragDetails?.localPosition;
+            } else if (notification is OverscrollNotification) {
+              dragPosition = notification.dragDetails?.localPosition;
+            } else {
+              dragPosition = null;
+            }
+            final windowSize = context.size;
+            setState(() {
+              showScrollLabels = windowSize != null &&
```

Comment:
This setState is too often.
We don't actually need to call it unless the value has changed.